pkg/scheduler/scheduler.go

```diff
@@ -377,23 +364,20 @@ func (s *Scheduler) nominate(ctx context.Context, workloads []workload.Info, sna
 }
 
 // resourcesToReserve calculates how much of the available resources in cq/cohort assignment should be reserved.
-func resourcesToReserve(e *entry, cq *cache.ClusterQueueSnapshot) resources.FlavorResourceQuantities {
+func resourcesToReserve(e *entry, cq *cache.ClusterQueueSnapshot) resources.FlavorResourceQuantitiesFlat {
 	if e.assignment.RepresentativeMode() != flavorassigner.Preempt {
 		return e.assignment.Usage
 	}
-	reservedUsage := make(resources.FlavorResourceQuantities)
-	for flavor, resourceUsage := range e.assignment.Usage {
-		reservedUsage[flavor] = make(map[corev1.ResourceName]int64)
-		for resource, usage := range resourceUsage {
-			cqQuota := cq.QuotaFor(resources.FlavorResource{Flavor: flavor, Resource: resource})
-			if !e.assignment.Borrowing {
-				reservedUsage[flavor][resource] = max(0, min(usage, cqQuota.Nominal-cq.Usage[flavor][resource]))
+	reservedUsage := make(resources.FlavorResourceQuantitiesFlat)
+	for fr, usage := range e.assignment.Usage {
+		cqQuota := cq.QuotaFor(fr)
+		if !e.assignment.Borrowing {
+			reservedUsage[fr] = max(0, min(usage, cqQuota.Nominal-cq.Usage[fr.Flavor][fr.Resource]))
+		} else {
+			if cqQuota.BorrowingLimit == nil {
+				reservedUsage[fr] = usage
 			} else {
-				if cqQuota.BorrowingLimit == nil {
-					reservedUsage[flavor][resource] = usage
-				} else {
-					reservedUsage[flavor][resource] = min(usage, cqQuota.Nominal+*cqQuota.BorrowingLimit-cq.Usage[flavor][resource])
-				}
+				reservedUsage[fr] = min(usage, cqQuota.Nominal+*cqQuota.BorrowingLimit-cq.Usage[fr.Flavor][fr.Resource])
 			}
 		}
```

Comment:
What about reversing the condition?

```go
if e.assignment.Borrowing {
	if cqQuota.BorrowingLimit == nil {
		reservedUsage[fr] = usage
	} else {
		reservedUsage[fr] = min(usage, cqQuota.Nominal+*cqQuota.BorrowingLimit-cq.Usage[fr.Flavor][fr.Resource])
	}
} else {
    reservedUsage[fr] = max(0, min(usage, cqQuota.Nominal-cq.Usage[fr.Flavor][fr.Resource]))
}
```

Reply:
Done. Also changed to `Usage.For` to reduce verbosity